sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -0,0 +1,985 @@
+
+            await using SearchIndexingBufferedSender<SimpleDocument> indexer =
+                client.CreateIndexingBufferedSender(
+                    new SearchIndexingBufferedSenderOptions<SimpleDocument>()
+                    {
+                        AutoFlush = false
+                    });
+            AssertNoFailures(indexer);
+            ConcurrentDictionary<int, IndexDocumentsAction<SimpleDocument>> pending = TrackPending(indexer);
+
+            await indexer.UploadDocumentsAsync(data);
+            await DelayAsync(EventDelay, EventDelay);
+            Assert.AreEqual(data.Length, pending.Count);
+
+            await indexer.FlushAsync();
+            await DelayAsync(EventDelay, EventDelay);
+            Assert.Zero(pending.Count);
+        }
+        #endregion
+
+        #region Dispose
+        [Test]
+        public async Task Dispose_Flushes()
+        {
+            await using SearchResources resources = await SearchResources.CreateWithEmptyIndexAsync<SimpleDocument>(this);
+            SearchClient client = resources.GetSearchClient();
+            SimpleDocument[] data = SimpleDocument.GetDocuments((int)(BatchSize * 1.5));
+
+            await using SearchIndexingBufferedSender<SimpleDocument> indexer =
+                client.CreateIndexingBufferedSender(
+                    new SearchIndexingBufferedSenderOptions<SimpleDocument>()
+                    {
+                        AutoFlush = false
+                    });
+            AssertNoFailures(indexer);
+            await indexer.UploadDocumentsAsync(data);
+
+            await ((IAsyncDisposable)indexer).DisposeAsync();
+            await WaitForDocumentCountAsync(resources.GetSearchClient(), data.Length);
+        }
+
+        [Test]
+        public async Task Dispose_Blocks()
+        {
+            await using SearchResources resources = await SearchResources.CreateWithEmptyIndexAsync<SimpleDocument>(this);
+            SearchClient client = resources.GetSearchClient();
+            SimpleDocument[] data = SimpleDocument.GetDocuments((int)(BatchSize * 1.5));
+
+            await using SearchIndexingBufferedSender<SimpleDocument> indexer =
+                client.CreateIndexingBufferedSender(
+                    new SearchIndexingBufferedSenderOptions<SimpleDocument>()
+                    {
+                        AutoFlush = false
+                    });
+            AssertNoFailures(indexer);
+            ConcurrentDictionary<int, IndexDocumentsAction<SimpleDocument>> pending = TrackPending(indexer);
+
+            await indexer.UploadDocumentsAsync(data);
+            await DelayAsync(EventDelay, EventDelay);
+            Assert.AreEqual(data.Length, pending.Count);
+
+            await ((IAsyncDisposable)indexer).DisposeAsync();
+            await WaitForDocumentCountAsync(resources.GetSearchClient(), data.Length);
+        }
+
+        [Test]
+        public async Task Dispose_UndisposedNoCrash()
```

Comment:
Should we check that a disposed instance throws at least on some methods, but not when double-disposed?

Reply:
Added tests for both cases